From 81cf08c5cdfe91b4e103a10d6fc74d8db6646d4d Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 28 Nov 2013 11:27:16 +0100
Subject: [PATCH 139/142] BUG/MAJOR: check: fix haproxy crash during
 soft-stop/soft-start

This is the continuation of previous fix bc16cd8 "BUG/MAJOR: fix haproxy
crash when using server tracking instead of checks", the soft-stop/start
states were not addressed by this fix.
---
 src/checks.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/checks.c b/src/checks.c
index 1cbc81e..d430c9b 100644
--- a/src/checks.c
+++ b/src/checks.c
@@ -576,7 +576,7 @@ static void set_server_disabled(struct check *check) {
 
 	if (s->state & SRV_CHECKED)
 		for(srv = s->tracknext; srv; srv = srv->tracknext)
-			set_server_disabled(check);
+			set_server_disabled(&srv->check);
 }
 
 static void set_server_enabled(struct check *check) {
@@ -610,7 +610,7 @@ static void set_server_enabled(struct check *check) {
 
 	if (s->state & SRV_CHECKED)
 		for(srv = s->tracknext; srv; srv = srv->tracknext)
-			set_server_enabled(check);
+			set_server_enabled(&srv->check);
 }
 
 static void check_failed(struct check *check)
-- 
1.8.1.5

